packages/domain/src/use-cases/authentication/Logout.ts

```diff
@@ -24,17 +41,25 @@ export interface ILogoutPresenter {
 
 export class Logout {
   constructor(
+    private sessionGateway: IRevokeSessionGateway,
     private walletGateway: IResettableWalletGateway,
     private credentialsGateway: IResettableCredentialsGateway,
+    private transactionGateway: IResettableTransactionGateway,
     private conversationsGateway: IConversationsGateway,
     private presenter: ILogoutPresenter,
   ) {}
 
   async execute(reason: LogoutReason): Promise<void> {
-    // TODO reset transactions gateway storage
+    const credentials = await this.credentialsGateway.getCredentials();
+
+    if (!credentials) {
+      never('User is not authenticated');
+    }
+
+    await this.sessionGateway.revoke({ authorizationId: credentials.authorizationId });
     await this.walletGateway.reset();
     await this.conversationsGateway.reset();
-
+    await this.transactionGateway.reset();
     await this.credentialsGateway.invalidate();
```

Comment:
I don't know if intentional or not, but with the `IRevokeSessionGateway` you are implicitly promoting "session" as a domain object.

Last time I worked on this area I thoughts that "credentials" became a poor abstraction over time (due to the changes we had to made to it). Session is a good replacement.

Replacing Credentials with Session is a bigger work for the scope of this ticket. So for now I would limit the revoke of credentials as implementation detail of `IResettableCredentialsGateway#invalidate`.